extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageAssetFetcherService.java

```diff
@@ -0,0 +1,47 @@
+package de.sovity.edc.ext.wrapper.api.ui.pages.transferhistory;
+
+import de.sovity.edc.ext.wrapper.api.common.model.AssetDto;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.connector.spi.asset.AssetService;
+import org.eclipse.edc.connector.spi.transferprocess.TransferProcessService;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcess;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.types.domain.asset.Asset;
+import org.jetbrains.annotations.NotNull;
+
+import static de.sovity.edc.ext.wrapper.utils.EdcDateUtils.utcMillisToOffsetDateTime;
+import static de.sovity.edc.ext.wrapper.utils.MapUtils.mapValues;
+
+@RequiredArgsConstructor
+public class TransferHistoryPageAssetFetcherService {
+    private final AssetService assetService;
+    private final TransferProcessService transferProcessService;
+
+    public AssetDto getAssetForTransferHistoryPage(String transferProcessId) {
+
+        var transferProcessById = transferProcessService.findById(transferProcessId);
+        if (transferProcessById == null) {
+            throw new EdcException("Could not find transfer process with ID %s.".formatted(transferProcessId));
+        }
+        return getAssetFromTransferProcess(transferProcessById);
+    }
+
+    @NotNull
+    private AssetDto getAssetFromTransferProcess(TransferProcess process) {
+        String assetId = process.getDataRequest().getAssetId();
+        var asset = assetService.findById(process.getDataRequest().getAssetId());
+        if (asset == null) {
+            asset = Asset.Builder.newInstance().id(assetId).build();
```

Comment:
May return an http 400 error

Reply:
Marking as resolved for now. However, Need to check with @richardtreier on how the errors would be handled by the api wrapper